components/nexusoperations/completion.go

```diff
@@ -127,11 +129,59 @@ func handleUnsuccessfulOperationError(
 	}
 }
 
+// Adds a NEXUS_OPERATION_STARTED history event and sets the operation state machine to NEXUS_OPERATION_STATE_STARTED.
+// Necessary if the completion is received before the start response.
+func fabricateStartedEventIfMissing(
+	node *hsm.Node,
+	requestID string,
+	operationID string,
+	startTime *timestamppb.Timestamp,
+	links []*commonpb.Link,
+) error {
+	data, err := hsm.MachineData[Operation](node)
+	if err != nil {
+		return err
+	}
+
+	if TransitionStarted.Possible(data) {
+		if operationID == "" {
+			return serviceerror.NewNotFound("operation not found")
```

Comment:
Not sure I understand why you're returning not found here.

Reply:
I did that because that's what we are returning in other places where a transition is not possible, but thinking about it more I'm not sure we even need to error here.